examples/threathunter/threat_intelligence/stix_taxii.py

```
@@ -0,0 +1,270 @@
+                logging.info(f"collection:{collection}; type data_set; breaking")
+                break
+            if num_reports == 0:
+                num_times_empty_content_blocks += 1
+            if num_times_empty_content_blocks > self.config.fail_limit:
+                logging.error('Max fail limit reached; Exiting.')
+                break
+            if reports_limit is not None:
+                reports_limit -= num_reports
+
+    def import_collections(self, available_collections):
+        desired_collections = self.config.collections
+        desired_collections = [x.strip()
+                               for x in desired_collections.lower().split(',')]
+
+        want_all = False
+        if '*' in desired_collections:
+            want_all = True
+
+        for collection in available_collections:
+            if collection.type != 'DATA_FEED' and collection.type != 'DATA_SET':
+                logging.debug(f"collection:{collection}; type not feed or data")
+                continue
+            if not collection.available:
+                logging.debug(f"collection:{collection} not available")
+                continue
+            if want_all or collection.name.lower() in desired_collections:
+                yield from self.import_collection(collection)
+
+    def generate_reports(self):   # NOTE:ignoring binary for now
+        reports = []
+
+        self.create_taxii_client()
+        if not self.client:
+            logging.error('Unable to create taxii client.  Exiting...')
+            return reports
+
+        available_collections = self.query_collections()
+        if not available_collections:
+            logging.warning('Unable to find any collections.  Exiting...')
+            return reports
+
+        reports = self.import_collections(available_collections)
+        if not reports:
+            logging.warning('Unable to import collections.  Exiting...')
+            return reports
+
+        return reports
```

error vs warning in the logs, but the same thing is returned? This seems like a place for fast fail on required fields instead of a just an error message vs a warning message.

---

With these three potential fail points, each results in a failure to send reports to CB. Since the user may have multiple ThreatIntel sources configured, we don't want them all to fail if one doesn't connect, doesn't have collections, or fails to import collections. 

I have updated the get_send_reports function (name tbd) to check if reports is empty. If it is, the code skips to the next ThreatIntel site in the configuration